src/annotator/integrations/pdf2htmlEx.tsx

```diff
@@ -0,0 +1,551 @@
+ */
+class BannerController implements Destroyable {
+  /** Top-level DOM element associated with the PDF.js viewer. */
+  private _pdfjsContainer: HTMLElement;
+  private _container: PreactContainer | null;
+  private _contentInfo: ContentInfoConfig | null;
+
+  /** Warning that the current PDF does not have selectable text. */
+  private _noTextWarning: boolean;
+
+  constructor() {
+    this._pdfjsContainer = document.querySelector(
+      '#outerContainer',
+    ) as HTMLElement;
+
+    this._contentInfo = null;
+    this._noTextWarning = false;
+    this._container = null;
+  }
+
+  /**
+   * Show a banner with information about the provider of the PDF.
+   *
+   * This is a contractual requirement for some LMS content providers.
+   */
+  setContentInfo(info: ContentInfoConfig) {
+    this._contentInfo = info;
+    this._update();
+  }
+
+  /**
+   * Set whether the "PDF has no selectable text" notice is shown.
+   */
+  showNoTextWarning(show: boolean) {
+    this._noTextWarning = show;
+    this._update();
+  }
+
+  destroy() {
+    this._container?.destroy();
+  }
+
+  private _update() {
+    const show = this._noTextWarning || this._contentInfo;
+    if (!show) {
+      this._container?.destroy();
+      this._container = null;
+
+      // Undo inline styles applied when the banner is shown. The banner will
+      // then gets its normal 100% height set by PDF.js's CSS.
+      this._pdfjsContainer.style.height = '';
+
+      return;
+    }
+
+    if (!this._container) {
+      this._container = new PreactContainer('banner', () => this._render());
+      document.body.prepend(this._container.element);
+    }
+
+    this._container.render();
+
+    // The `#outerContainer` element normally has height set to 100% of the body.
+    //
+    // Reduce this by the height of the banner so that it doesn't extend beyond
+    // the bottom of the viewport.
+    //
+    // We don't currently handle the height of the banner changing here.
+    const bannerHeight = this._container.element.getBoundingClientRect().height;
+    this._pdfjsContainer.style.height = `calc(100% - ${bannerHeight}px)`;
+  }
+
+  private _render() {
+    return (
+      <Banners>
+        {this._contentInfo && <ContentInfoBanner info={this._contentInfo} />}
+        {this._noTextWarning && <WarningBanner />}
+      </Banners>
+    );
+  }
+}
+
+/**
+ * Integration that works with PDF.js
+ */
+export class PDF2HtmlExIntegration extends TinyEmitter implements Integration {
+  private _annotator: Annotator;
+
+  /** Banners shown at the top of the PDF viewer. */
+  private _banner: BannerController;
+
+  /**
+   * A flag that indicates whether `destroy` has been called. Used to handle
+   * `destroy` being called during async code elsewhere in the class.
+   */
+  private _destroyed: boolean;
+  private _listeners: ListenerCollection;
+  private _observer: MutationObserver;
+  private _pdfContainer: HTMLElement;
+  // private _pdfMetadata: PDFMetadata;
```

Comment:
Can be removed, the `HTMLMetadata` is more appropriate for the pdf2HtmlEx use case.